acceptance_tests/app_endpoint.go

```
@@ -0,0 +1,111 @@
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/app_endpoints"
+)
+
+func createAppEndpoint(ctx context.Context, client *api.Client) error {
+	appEndpointRequest := app_endpoints.AppEndpointRequest{
+		Name:             globalAppEndpointName,
+		Bucket:           globalBucketName,
+		UserXattrKey:     "syncFnXattr",
+		DeltaSyncEnabled: true,
+		Scopes: app_endpoints.Scopes{
+			"_default": app_endpoints.Scope{
+				Collections: map[string]app_endpoints.Collection{
+					"_default": app_endpoints.Collection{
+						AccessControlFunction: "function (doc, oldDoc, meta) {return true;}",
+						ImportFilter:          "function (doc) {return true;}",
+					},
+				},
+			},
+		},
+	}
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints",
+		globalHost,
+		globalOrgId,
+		globalProjectId,
+		globalClusterId,
+		globalAppServiceId,
+	)
+	cfg := api.EndpointCfg{
+		Url:           url,
+		Method:        http.MethodPost,
+		SuccessStatus: http.StatusCreated,
+	}
+
+	_, err := client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		appEndpointRequest,
+		globalToken,
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func appEndpointWait(ctx context.Context, client *api.Client) error {
+	const maxWaitTime = 10 * time.Minute
+	const checkInterval = 1 * time.Minute
+
+	deadline := time.Now().Add(maxWaitTime)
+
+	for time.Now().Before(deadline) {
+		url := fmt.Sprintf(
+			"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s",
+			globalHost,
+			globalOrgId,
+			globalProjectId,
+			globalClusterId,
+			globalAppServiceId,
+			globalAppEndpointName,
+		)
+
+		cfg := api.EndpointCfg{
+			Url:           url,
+			Method:        http.MethodGet,
+			SuccessStatus: http.StatusOK,
+		}
+
+		response, err := client.ExecuteWithRetry(
```

When this fails for reasons other than "resource not found", the error is ignored and the loop continues.
I would suggest to Log unexpected errors.  Also, should we abort after repeated failures instead of looping?

---

The last error is returned if the retry loop exhausts its retries